java-checks/src/main/java/org/sonar/java/checks/IgnoredTestsCheck.java

```diff
@@ -53,24 +53,35 @@ public void visitNode(Tree tree) {
     }
     MethodTree methodTree = (MethodTree) tree;
     SymbolMetadata symbolMetadata = methodTree.symbol().metadata();
-    if (isSilentlyIgnored(symbolMetadata, "org.junit.Ignore") || isSilentlyIgnored(symbolMetadata, "org.junit.jupiter.api.Disabled")) {
-      reportIssue(methodTree.simpleName(), "Fix or remove this skipped unit test");
+
+    // check for @Ignore or @Disabled annotations
+    boolean hasIgnoreAnnotation = isSilentlyIgnored(symbolMetadata, "org.junit.Ignore");
+    boolean hasDisabledAnnotation = isSilentlyIgnored(symbolMetadata, "org.junit.jupiter.api.Disabled");
+    if (hasIgnoreAnnotation || hasDisabledAnnotation) {
+      reportIssue(methodTree.simpleName(), String.format("Either add an explanation about why this test is skipped or remove the " +
+        "\"@%s\" annotation.", hasIgnoreAnnotation ? "Ignore" : "Disabled"));
     }
+
+    // check for "assumeFalse(true)" and "assumeTrue(false)"-calls, which may also result in permanent skipping of the given test
     BlockTree block = methodTree.block();
-    if(block != null) {
+    if (block != null) {
       block.body().stream()
         .filter(s -> s.is(Tree.Kind.EXPRESSION_STATEMENT))
         .map(s -> ((ExpressionStatementTree) s).expression())
         .filter(s -> s.is(Tree.Kind.METHOD_INVOCATION))
         .map(MethodInvocationTree.class::cast)
         .filter(ASSUME_METHODS::matches)
         .filter(IgnoredTestsCheck::hasConstantOppositeArg)
-        .forEach(mit -> reportIssue(mit.methodSelect(), "Fix or remove this skipped unit test"));
+        .forEach(mit -> reportIssue(mit, "Either remove this assumption or use an @Ignore or @Disabled annotation in combination with " +
```

Comment:
My reasoning behind changing the part being highlighted was, that since it is not in itself problematic to use `assumeTrue(...)` and only becomes problematic with a constant boolean value, the argument should be part of the highlighted code, given that it is a defining part of the problem. You make a good point, though, in that it could interfere with usability when the argument becomes complex. So generally speaking we rather want to opt for simpler highlights, even if that could mean we potentially don't highlight all parts of the problematic code?

Reply:
Good point, I agree that the argument is part of the problem, the current situation could be improved.

> So generally speaking we rather want to opt for simpler highlights, even if that could mean we potentially don't highlight all parts of the problematic code?

I think we should aim for both (as much as possible..)! To do this, you can use secondary locations. You can have a look at #2914 for an example!

In this case, I would change the message to state that the assume is called with a constant argument and use a secondary location to highlight the argument. The secondary location can still be huge, but it will not be shown when browsing the file, only the main location will be.

Reply:
Perfect, that sounds like exactly the fitting solution in this case, thanks!